manifests/rule.pp

```diff
@@ -65,7 +65,7 @@
 #
 define logrotate::rule (
   Array[String]                                       $log_files,
-  Boolean                                             $compress                  = true,
+  Boolean                                             $compress                  = $logrotate::compress,
```

Comment:
@jeannegreulich This won't work.

There is no guarantee that the `logrotate` class has been included prior to the call to `logrotate::rule`.

You'll need to set them to `undef` and then perform the logic in the code.

Reply:
@trevor-vaughan The init log rotate is needed prior to the rule because it is what includes the logrotate.d where the logrotate::rules are put.  I included it to make sure it was there prior to the rule.  It seemed the best way to do it.  It allow the users to set a new default for all rules or change them individually.  If you don't want that I can remove it and and hard code the defaults again.   I tried setting them to undefined and not including them in the sub rules but the they did not pick up the defaults from the logrotate.conf file.

Reply:
@jeannegreulich ensure the logrotate rule directory here and wrap both instances in `if !defined(logrotate rule directory) { ensure logrotate rule directory }`